components/admin_console/brand_image_setting/brand_image_setting.jsx

```diff
@@ -62,86 +70,129 @@ export default class BrandImageSetting extends React.PureComponent {
 
             reader.readAsDataURL(this.state.brandImage);
         }
+
+        if (this.props.saving && this.state.brandImage && !this.props.disabled) {
+            this.handleImageSubmit();
```

Comment:
Yes, I think here, the "cleanest" approach is to use refs. I you reference a node which is a component, you get the instance of the component and you can call the internal methods of the component itself, so, I would add a ref in the parent to the instance of this component and then call. this.brandImageSettingRef.current.saveImage().

The callback to register a callback can work, but sounds complicated.

Use methods of a component from the outside is rare and normally unexpected for the developers because is not the common use case, so I would suggest to add a comment in the function like `// This function is used from the outside to trigger the save of the image when needed.`